climada/engine/impact.py

```diff
@@ -1736,6 +1737,48 @@ def stack_attribute(attr_name: str) -> np.ndarray:
             **kwargs,
         )
 
+    def assign_centroids(self, hazard, distance='euclidean',
+                         threshold=u_coord.NEAREST_NEIGHBOR_THRESHOLD):
+        """
+        Assign for each impact coordinate closest hazard coordinate.
+        Adds a new attribute to the impact object, ...
+        
+        Uses ``u_coord.assign_centroids_to_gdf()``. See there for details and parameters
+
+        Parameters
+        ----------
+        hazard : Hazard
+            Hazard to match (with raster or vector centroids).
+        distance : str, optional
+            Distance to use in case of vector centroids.
+            Possible values are "euclidean", "haversine" and "approx".
+            Default: "euclidean"
+        threshold : float
+            If the distance (in km) to the nearest neighbor exceeds `threshold`,
+            the index `-1` is assigned.
+            Set `threshold` to 0, to disable nearest neighbor matching.
+            Default: 100 (km)
+        """
+
+        haz_type = hazard.tag.haz_type
+        centr_haz = 'centr_' + haz_type
+
+        #Assert that the imp crs is epsg:4326, as it is required by the u_coord methods
+        if not u_coord.equal_crs(self.crs, 'EPSG:4326'):
+            raise ValueError('Set Impact to lat/lon crs (EPSG:4326)!')
```

Comment:
One more question (sorry 😁) - which u_coord methods require epsg:4326?
And why can't they speak for themselves, i.e. raise an exception if the crs is wrong?